connect/mirror/src/main/java/org/apache/kafka/connect/mirror/OffsetSyncStore.java

```diff
@@ -105,7 +106,10 @@ private KafkaBasedLog<byte[], byte[]> createBackingStore(MirrorCheckpointConfig
     /**
      * Start the OffsetSyncStore, blocking until all previous Offset Syncs have been read from backing storage.
      */
-    public void start() {
+    public void start(boolean initializationMustReadToEnd) {
+        this.initializationMustReadToEnd = initializationMustReadToEnd;
+        log.info("OffsetSyncStore initializationMustReadToEnd:{}{}", initializationMustReadToEnd,
+                initializationMustReadToEnd ? " - fewer checkpoints may be emitted" : "");
```

Comment:
nit: Make this more verbose and user-oriented. They don't care that the variable is called initializationMustReadToEnd, and "Must read to end" is a very technical description of what is happening here.
Specify more precisely which checkpoints aren't being emitted. Fewer could mean every other one, but it's actually offsets which were mirrored before the task started.

Actually, this message works better if you put it after the backingStore.start() call: You can print out the oldest offset sync to say that translation is starting there, and whether this is limited by the initialization setting.

Reply:
thanks we reworded the message where we catch the error, though we didn't write the offsets as there may be many groups handled by a task